app/models.py

```diff
@@ -64,14 +73,30 @@ def check_password(self, password):
     db.Model.metadata,
     db.Column('user_id', UUID(as_uuid=True), db.ForeignKey('users.id')),
     db.Column('service_id', UUID(as_uuid=True), db.ForeignKey('services.id')),
-
     UniqueConstraint('user_id', 'service_id', name='uix_user_to_service')
 
 )
 
 
-class Service(db.Model):
+class ServiceHistory(db.Model):
```

Comment:
Most of the fields are nullable, the main difference is the composite primary key of id and version, so I think inheritance wouldn't be the way to go for this model.